File: crypto-app-spring/investment-service/src/main/java/com/ironhack/investmentservice/model/Investment.java
package com.ironhack.investmentservice.model;

import org.apache.catalina.*;

import javax.persistence.*;
import java.math.*;
import java.time.*;

@Entity
public class Investment {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private Long user;
    private BigDecimal quantity;
    private String coin;
    private LocalDate dateInvestment;


    public Investment() {
    }

    public Investment(Long user, BigDecimal quantity, String coin, LocalDate dateInvestment) {
        this.user = user;
        this.quantity = quantity;
        this.coin = coin;
        this.dateInvestment = dateInvestment;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getUser() {
        return user;
    }

    public void setUser(Long user) {
        this.user = user;
    }

    public BigDecimal getQuantity() {
        return quantity;
    }

    public void setQuantity(BigDecimal quantity) {
        this.quantity = quantity;
    }

    public String getCoin() {
        return coin;
    }

    public void setCoin(String coin) {
        this.coin = coin;
    }

    public LocalDate getDateInvestment() {
        return dateInvestment;
    }

    public void setDateInvestment(LocalDate dateInvestment) {
        this.dateInvestment = dateInvestment;
    }
}

File: crypto-app/src/app/models/investment.ts
export class Investment {

    constructor(
        private _id: number,
        private _user: number,
        private _quantity: number,
        private _coin: string,
        private _dateInvestment: Date,

    ) {}

    public get dateInvestment(): Date {
        return this._dateInvestment;
    }
    public set dateInvestment(value: Date) {
        this._dateInvestment = value;
    }
    public get coin(): string {
        return this._coin;
    }
    public set coin(value: string) {
        this._coin = value;
    }
    public get quantity(): number {
        return this._quantity;
    }
    public set quantity(value: number) {
        this._quantity = value;
    }
    public get user(): number {
        return this._user;
    }
    public set user(value: number) {
        this._user = value;
    }
    public get id(): number {
        return this._id;
    }
    public set id(value: number) {
        this._id = value;
    }

}
